gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -872,6 +870,78 @@ public void deleteObjects(List<StorageResourceId> fullObjectNames) throws IOExce
     }
   }
 
+  private void createGrpcStubIfEnabled(GoogleCloudStorageOptions options, Credentials credentials) {
+    if (!this.storageOptions.isGrpcEnabled()) {
+      return;
+    }
+
+    this.watchdog =
+        Watchdog.create(Duration.ofMillis(options.getGrpcMessageTimeoutCheckInterval()));
```

Comment:
This is a non-obvious side effect for create method, can we move `watchdog` inside the gRPC stub object, for example?